src/disk_io_thread_pool.cpp

```diff
@@ -142,7 +142,6 @@ namespace libtorrent {
 		// but do it to avoid acquiring the mutex in the trivial case
 		if (m_num_idle_threads >= queue_size) return;
 		std::lock_guard<std::mutex> l(m_mutex);
-		if (m_abort) return;
```

Comment:
even though we're aborting, we still need to service disk jobs, as long as there are any in the queue